### PR TITLE
Add Timeout, RetryCount to APIArg

### PR DESCRIPTION
**Pull request by patrickxb**
See https://keybase.atlassian.net/browse/CORE-3300 for discussion of this ticket.

r? @oconnor663 or @mmaxim

### CODE REVIEW

**Comment by patrickxb** [comment]
retest this please

File: go/libkb/api.go [hunk 1]
```diff
@@ -262,6 +266,76 @@ func doRequestShared(api Requester, arg APIArg, req *http.Request, wantJSONRes b
...
+	}
+	c := make(chan response, 1)
+	go func() {
+		var r response
+		r.resp, r.err = cli.cli.Do(req)
+		c <- r
+	}()
+
+	select {
+	case <-ctx.Done():
```
Comment (mmaxim) [inline go/libkb/api.go]: Is the only reason not to use a timer here related to the TODO comment above?

Comment (patrickxb) [inline go/libkb/api.go]: yeah, basically.

also, go 1.7 net/http has context built in, so this will prepare us for that as well.

Comment (mmaxim) [inline go/libkb/api.go]: Ah nice!

**Comment by mmaxim** [comment]
👍

**Comment by patrickxb** [comment]
you guys good with the timeouts I chose?  5s with 4 attempts for sesscheck, favorites list.  Current default is 60s, 1 attempt.

**Comment by maxtaco** [comment]
Does it make sense to back off a bit too?  If the timeouts are due to the
server being over capacity, it might make sense to retry faster but then
back off faster.

**Comment by patrickxb** [comment]
I thought of that.  Ok, I'll add in a backoff.

**Comment by patrickxb** [comment]
@maxtaco @mmaxim when you get a chance

**Comment by mmaxim** [comment]
LGTM!

**Pull request merged**